fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MvUtils.java

```diff
@@ -581,6 +583,76 @@ public static boolean isRedundantPredicate(ScalarOperator predicate) {
         return predicate.isPushdown() || predicate.isRedundant();
     }
 
+    public static Set<ScalarOperator> getPredicateForRewrite(OptExpression root) {
```

Comment:
This function only extracts predicates from join and scan node which scan node will exclude `IsNullPredicateOperator ` predicates if it's not null and its column ref is in the join's on predicates?

Am I right? may add more docs for this function?

Reply:
yes. for inner join and semi join. I will add it.